src/cmd/sanity_test/sanity_tests.go

```diff
@@ -323,91 +328,120 @@ func checkOnedriveRestoration(
 			fileSizes[itemName] = ptr.Val(driveItem.GetSize())
 		}
 
-		folderPermission[itemID] = permissionsIn(ctx, client, driveID, itemID, folderPermission[itemID])
+		if driveItem.GetFolder() == nil && driveItem.GetPackage() == nil {
+			continue
+		}
+
+		// currently we don't restore blank folders.
+		// skip permission check for empty folders
+		if ptr.Val(driveItem.GetFolder().GetChildCount()) == 0 {
+			fmt.Println("skipped empty folder: ", itemName)
+			continue
+		}
+
+		permissionIn(ctx, client, driveID, itemID, itemName, folderPermission)
+		getOneDriveChildFolder(ctx, client, driveID, itemID, itemName, fileSizes, folderPermission, startTime)
+	}
+
+	getRestoreData(ctx, client, *drive.GetId(), restoreFolderID, restoreFile, restoreFolderPermission, startTime)
+
+	for checkFolderName, checkfolderPer := range folderPermission {
```

Comment:
Nit: I think this will be more readable.
```suggestion
	for folderName, permissions := range folderPermission {
```